README.md

```diff
@@ -16,46 +16,93 @@ Great for:
+
 
 What's with the name?
 --------------------
 We wanted an installation of Kubernetes that was half the size in terms of memory footprint. Kubernetes is a
 10 letter word stylized as k8s. So something half as big as Kubernetes would be a 5 letter word stylized as
-k3s. There is no long form of k3s and no official pronunciation.
+K3s. There is no long form of K3s and no official pronunciation.
+
+
+Is this a fork?
+---------------
+No, it's a distribution. A fork implies continued divergence from the original. This is not K3s's goal or practice. K3s explicitly intends to not change any core Kubernetes functionality. We seek to remain as close to upstream Kubernetes as possible. We do maintain a small set of patches (well under 1000 lines) important to K3s's usecase and deployment model. We maintain patches for other components as well. When possible, we contribute these changes back to the upstream projects, for example with [SELinux support in containerd](https://github.com/containerd/cri/pull/1487/commits/24209b91bf361e131478d15cfea1ab05694dc3eb). This is a common practice amongst software distributions.
+
+K3s is a distribution because it packages additional components and services necessary for a fully functional cluster that go beyond vanilla Kubernetes. These are opinionated choices on technologies for components like ingress, storage class, network policy, service load balancer, and even container runtime. These choices and technologies are touched on in more detail in the [What is this?](what-is-this) section.
+
+
+How is this lightweight or smaller than upstream Kubernetes?
+---
+There are two major ways that K3s is lighter weight than upstream Kubernetes:
+1. The memory footprint to run it is smaller
+2. The binary, which contains all the non-containerized components needed to run a cluster, is smaller
+
+The memory footprint is reduced primarily by running many components inside of single process. This eliminates significant overhead that would otherwise be duplicated for each component.
+
+The binary is smaller primarily by removing some unused Kubernetes code.
+
+What have you removed from upstream Kubernetes?
+---
+This is a common point of confusion because it has changed over time. Early versions of K3s had much more removed than current version. Current versions of K3s remove two things:
+1. In-tree storage drivers
+2. In-tree cloud provider
+
+We remove these to achieve a smaller binary size. They can be removed while remaining conformant because neither affect core Kubernetes functionality. They are also dependent on cloud or data center technologies/services, which are generally not available in K3s's usecases. Finally, both have out-of-tree alternatives in the form of [CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md) and [CCM](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/), which work in K3s and which upstream is moving towards.
```

Comment:
The crucial point about tailored support through these out-of-tree components feesl like it could get lost in this paragraph.